backend/src/main/java/gov/cdc/usds/simplereport/db/repository/EternalSystemManagedEntityRepository.java

```diff
@@ -0,0 +1,33 @@
+package gov.cdc.usds.simplereport.db.repository;
+
+import java.util.List;
+import java.util.UUID;
+
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.NoRepositoryBean;
+import org.springframework.data.repository.CrudRepository;
+
+import gov.cdc.usds.simplereport.db.model.EternalSystemManagedEntity;
+
+/**
+ * Base repository interface for soft-deletable entities. Should implement all CRUD operations, but with
+ * restrictions against unintentionally retrieving soft-deleted items.
+ *
+ * @param <T> a soft-delete-only subclass of {@link EternalSystemManagedEntity}.
+ */
+@NoRepositoryBean
+public interface EternalSystemManagedEntityRepository<T extends EternalSystemManagedEntity> extends CrudRepository<T, UUID> {
```

Comment:
This is a sort of frustrating interface. I'd say we should just use these two methods and a constant as a mix-in, but there might be complications if I'm right about the audit fields not being set correctly in `delete(T)`... 😕 

On the other hand, this also quietly brings in the rest of CrudRepository into ApiUserRepository, which is a little unexpected, though probably no more dangerous for ApiUserRepository than for any of the others (which is to say, a little more dangerous than I would like).

Reply:
can you spell out what's frustrating? and what you mean by mix-in? do you mean not-inherited?

Reply:
to the concerns about https://github.com/CDCgov/prime-simplereport/issues/294 , am I correct in saying that those concerns by definition do *not* apply to this interface? if they fail to capture audit information, what does it matter here, since this repo concerns tables that are non-audited?

Reply:
You are correct, yes. To spell it out a bit more fully:

* this interface is frustrating because it is word-for-word identical to another interface, other than the inheritance difference required to make it fit into our repository type tree (which is irrelevant to the functioning of this interface, other than making it show up in the right place). So to fit this feature into our repository type tree, we have to have two copies of it, which is, as I say, frustrating.
* we could conceivably turn this interface into a mix-in (though we might want to attach it to a base class on the Audited side anyway), allowing us to use multiple interface inheritance to get this working on both sides of the audited/system divide while maintaining only one copy of the code.
* _but_ if #294 is in fact a significant issue (yet to be verified, but we're now beginning to use the soft-delete functions enough that we really ought to do it), we would then need to change something about the EternalAuditedEntityRepository to cover that case, thus messing up the opportunity for code re-use. Which would, again, be frustrating.

Reply:
discussed offline.